mantidproject/mantid
# Build documentation in vanilla Python instead of MantidPlot

**Description of work.**

Switches the `docs-` build targets to use the mantidpython wrappers rather than MantidPlot. This required ensuring the screenshot tools could cope with having to start a QApplication object themselves. There has also been some general clean ups in the directives along with some new CMake options to control the docs build. The aim of these options is to reduce the time it takes to build the documentation. A full build took ~15min on my Ubuntu 18.04 box with decent NVMe drive. The options disable various directives, in particular `:plot:`, so that previewing documentation changes does not require a long build time. With all new options set to off the build time is ~1.5min on the same machine. 

**To test:**

Look at the documentation changes and check they make sense. Try out the new CMake options and make sure you can still build the documentation.

Fixes #28647

*This does not require release notes* because **as it is an internal change.**



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
